sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/text/TextTableProvider.java

```diff
@@ -141,6 +156,51 @@ public LinesReadConverter() {}
     }
   }
 
+  /** Read-side converter for {@link TextJsonTable} with format {@code 'json'}. */
+  @VisibleForTesting
+  static class JsonToRow extends PTransform<PCollection<String>, PCollection<Row>>
+      implements Serializable {
+
+    private ObjectMapper objectMapper;
+    private Schema schema;
+
+    @VisibleForTesting
+    public JsonToRow(ObjectMapper objectMapper, Schema schema) {
+      this.objectMapper = objectMapper;
+      this.schema = schema;
+    }
+
+    @Override
+    public PCollection<Row> expand(PCollection<String> input) {
+      return input
+          .apply(
+              ParDo.of(
+                  new DoFn<String, Row>() {
+                    @ProcessElement
+                    public void processElement(ProcessContext context) {
```

Comment:
Will need a design or documentation in the case if there is a row that does not match the schema. If you check pubsub json support, it supports a dead letter queue that sends rows to that queue if those rows does not match with the schema. You might reuse the same design.

Reply:
I may plan to create the file specifically for all rows that don't match the scheme, which is pretty similar to your design.

Do you think we really need a doc for that. Would you mind pointing me to how a design is filed?